lib/omniauth/strategies/pam.rb

```diff
@@ -38,21 +38,20 @@ def callback_phase
       end
 
       info do
-        info = { :nickname => uid, :name => uid }
-        info[:email] = "#{uid}@#{options[:email_domain]}" if options.has_key?(:email_domain)
-        info.merge!(parse_gecos || {})
-      end
+        info = { name: uid }
 
-      private
+        list_env = Rpam2.listenv(options[:service],
+                                 uid,
+                                 request["password"],
+                                 false)
```

Comment:
I'm finding that this is always returning an empty hash on a standard Ubuntu system. Previously, we were extracting the user's name from `gecos` (where, if they'd set it, could also have provided `location`, `phone`, `home_phone`, `description`).

What's different here with `Rpam2`? Do other things need to be installed on the system for this to work correctly?

Reply:
yes, I forgot to document because the tool was not built then:
https://github.com/devkral/pam_email_extractor

Reply:
but you can use everything which injects "email" in your pam environment